src/hooks/useOutsideClick/index.ts

```diff
@@ -0,0 +1,31 @@
+import { useEffect } from 'react';
+
+import type { AnyEvent, OutsideClickProps } from './types';
+
+const useOutsideClick: OutsideClickProps = (ref, handler) => {
+  useEffect(() => {
+    const listener = (event: AnyEvent) => {
+      const el = ref?.current;
+
+      /**
+       * Do nothing if clicking ref's
+       * element or descendent elements.
+       */
+      if (! el || el.contains(event.target as Node)) {
+        return;
+      }
+
+      handler(event);
+    };
+
+    document.addEventListener('mousedown', listener);
+    document.addEventListener('touchstart', listener);
+
+    return () => {
+      document.removeEventListener('mousedown', listener);
+      document.removeEventListener('touchstart', listener);
+    };
+  }, [ref, handler]);
+};
+
+export default useOutsideClick;
```

Comment:
duplicate hooks, please look inside `/hooks`, it already exists
please just use the single file hook, no need to separate types for such small amount of code.

careful when merging upstream `main` with old branch

![image](https://user-images.githubusercontent.com/27584221/197780956-4342b8a1-8b08-4236-99bb-f83538396446.png)